ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -83,177 +213,171 @@ export class ImportedDocuments extends React.Component {
+          record.entry_recid.includes(searchText) ||
+          record.output_pid?.includes(searchText) ||
+          record.document?.title.includes(searchText) ||
+          record.eitem?.output_pid?.includes(searchText) ||
+          record.series?.filter(serie => serie.output_pid?.includes(searchText))
+            .length > 0
+      );
+      console.log('Tick');
+      this.setState({
+        finalFilteredRecords: searchRecords,
+      });
+    }, 60);
+    return deb(searchText);
+  };
+
+  filteredRecords = () => {
+    const { statistics, data, selectedResult } = this.state;
+    const selectedFilterFunc = statistics[selectedResult].filterFunction;
+    const records = data.records;
+
+    const filteredRecords = records.filter(record =>
+      selectedFilterFunc(record)
     );
+    this.searchBarFilter(filteredRecords);
+    this.setActivePage(1);
   };
 
   renderCancelButton = () => {
-    const { data } = this.state;
+    const { isLoading } = this.state;
     const { taskId } = this.props;
 
     return (
-      <>
-        {!_isEmpty(data) && data.status === 'RUNNING' && (
-          <CancelImportTask logId={taskId} />
-        )}
-      </>
+      isLoading && (
+        <>
+          <CancelImportTask logId={taskId} />{' '}
+          <Icon loading name="circle notch" />
+          This may take a while. You may leave the page, the process will
+          continue in background. ImportedSearch
+        </>
+      )
     );
   };
 
   render() {
-    const { data } = this.state;
+    const {
+      data,
+      statistics,
+      selectedResult,
+      finalFilteredRecords,
+    } = this.state;
     return (
       <>
         {this.renderImportReportHeader()}
         {!_isEmpty(data) && data.status !== 'FAILED' ? (
           <>
-            <Header as="h3">Import report</Header>
-            {modeFormatter(data.mode)}{' '}
-            {!_isEmpty(data) ? (
-              (data.loaded_entries || data.loaded_entries === 0) &&
-              data.entries_count ? (
-                <span>
-                  {'Processed ' +
-                    data.loaded_entries +
-                    ' records out of ' +
-                    data.entries_count +
-                    '.'}
-                </span>
-              ) : (
-                <span>Processing file...</span>
+            <Segment>
+              {!_isEmpty(data) ? (
+                (data.loaded_entries || data.loaded_entries === 0) &&
+                data.entries_count ? (
+                  <Grid className="middle aligned">
+                    <Grid.Column width={3}>
+                      <Segment>
+                        Mode: {this.mode}
+                        <Divider />
+                        Status: {this.status}
+                      </Segment>
+                    </Grid.Column>
+                    <Grid.Column width={13}>
+                      <Statistic.Group widths="seven">
+                        {Object.keys(statistics).map(function(
```

Comment:
could we improve the readability of this part ? maybe makes sense to have a function component for this?